librz/include/rz_type.h

```diff
@@ -0,0 +1,185 @@
+// SPDX-FileCopyrightText: 2021 Anton Kochkov <anton.kochkov@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef RZ_TYPE_H
+#define RZ_TYPE_H
+
+#include <rz_types.h>
+#include <rz_util/rz_num.h>
+#include <rz_util/rz_print.h>
+#include <rz_bind.h>
+#include <rz_io.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+RZ_LIB_VERSION_HEADER(rz_type);
+
+typedef struct rz_type_target_t {
+	const char *cpu;
+	int bits;
+	const char *os;
+	bool big_endian;
+} RzTypeTarget;
+
+typedef struct rz_type_t {
+	void *user;
+	Sdb *sdb_types;
+	Sdb *formats; // for `pf` formats
+	RzTypeTarget *target;
+	RNum *num;
+	RzIOBind iob; // for RzIO in formats
+} RzType;
```

Comment:
What about calling this `RzTypeDB` or `RzTypeDatabase`? Then we can also later merge `RzBaseType` and `RzCType` to just `RzType`.

Reply:
Yeah, that would make more sense.

Reply:
+1 on RzTypeDatabase/RzTypeDb

Reply:
My favorite is `RzTypeDB` I think. Then functions like `rz_type_db_get_something` don't get too long and "db" is a very common abbreviation.

Reply:
Done, renamed to `RzTypeDB`, also renamed corresponding functions to `rz_type_db_*` where it really uses the instance as the database.